src/api.ts

```diff
@@ -66,59 +60,37 @@ export class API {
 
     /**
      * Fetch homepage doujins
+     * @param options Search Options
      */
-    fetchHomepage(page: string | number = 1, sort = SortMethods.RECENT): Promise<SearchResult> {
-        return this.search('*', page, sort);
+    fetchHomepage(options: APISearchOptions = {}): Promise<SearchResult> {
+        return this.search('*', options);
     }
 
     /**
      * Search by string query
+     * @param options Search Options
      */
-    async search(query: string, page: string | number = 1, sort = SortMethods.RECENT): Promise<SearchResult> {
-        page = Number(page);
-
-        if (isNaN(page)) throw new TypeError('page is not a number');
-        if (!SortValues.includes(sort)) throw new TypeError('sort method is not valid');
-
-        const res = await this.fetch(`/galleries/search?query=${query}&page=${page}&sort=${sort}`);
+    async search(query: string, options: APISearchOptions = {}): Promise<SearchResult> {
+        const res = await this.fetch(`/galleries/search?query=${query}${options.language ? ` ${options.language}` : ''}&page=${options.page ? `${options.page}` : 1}&sort=${options.sort ? `${options.sort}` : SortMethods.RECENT}`);
         return new SearchResult(res as APISearchResult);
     }
 
     /**
      * Search doujins with this tag
      * @param id Tag ID
-     * @param page Page number
-     * @param sort API sort method
+     * @param options Search Options
      */
-    async searchByTagID(
-        id: number | string,
-        page: string | number = 1,
-        sort = SortMethods.RECENT
-    ): Promise<SearchResult> {
-        id = Number(id);
-        page = Number(page);
-
-        if (isNaN(id)) throw new TypeError('tagId is not a number');
-        if (isNaN(page)) throw new TypeError('page is not a number');
-        if (!SortValues.includes(sort)) throw new TypeError('sort method is not valid');
```

Comment:
Why remove argument type checking on all methods? Especially when you pass an invalid `sort` param here it returns an api error thats nearly impossible to debug.